File: Back/src/main/java/com/example/buensaborback/domain/entities/Usuario.java
package com.example.buensaborback.domain.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Usuario extends Base{
    private String nombreUsuario;
    private String password;
    @ManyToOne
    @JoinColumn(name = "idRol")
    private Rol rol;

    public Usuario(String nombreUsuario, String password) {
        this.nombreUsuario = nombreUsuario;
        this.password = encryptPassword(password);
    }

    private String encryptPassword(String password) {
        PasswordEncoder encoder = new BCryptPasswordEncoder();
        return encoder.encode(password);
    }
}

File: Front/buenSaborFront/src/models/Usuario.ts
import Cliente, { ClienteRegistro } from "./Cliente";
import Rol from "./Rol";

export default class Usuario {
    id?: number;
    nombreUsuario?: string;
    //password?: string = "";
    rol: Rol = new Rol();
}

export class UsuarioLogin {
    nombreUsuario?: string;
    password: string = "";
}

export class UsaurioRegistro {
    id?: number;
    nombreUsuario?: string;
    rol: Rol = new Rol();
    cliente: ClienteRegistro = new ClienteRegistro();
}

export class UsuarioCliente {
    id?: number;
    nombreUsuario?: string;
    rol: Rol = new Rol();
    cliente: Cliente = new Cliente();
}